data-serving/data-service/src/model/location.ts

```diff
@@ -0,0 +1,82 @@
+import mongoose from 'mongoose';
+
+const fieldRequiredValidator = [
+    function (this: LocationDocument): boolean {
+        return (
+            this != null &&
+            this.country == null &&
+            this.administrativeAreaLevel1 == null &&
+            this.administrativeAreaLevel2 == null &&
+            this.locality == null
+        );
+    },
+    'One of location.country, location.administrativeRegion1, ' +
+        'location.administrativeRegion2, or locality is required',
```

Comment:
Which schema do these prefixes refer to (i.e. why isn't locality prefixed with `location.`)?

Reply:
because i am bad at typinggg -- going to remove the prefixes, it's confusing